src/main/java/store/model/service/OrderService.java

```diff
@@ -0,0 +1,72 @@
+package store.model.service;
+
+import java.util.ArrayList;
+import java.util.List;
+import store.model.domain.Cart;
+import store.model.domain.Product;
+import store.model.domain.Stock;
+
+public class OrderService {
+    private final Cart cart;
+    private String name;
+    private int quantity;
+
+    public OrderService() {
+        this.cart = new Cart();
+    }
+
+    public Cart order(Stock stock, String requestProducts) {
+        List<String> products = new ArrayList<>(List.of(requestProducts.trim().split(",")));
+
+        for (String product : products) {
+            orderParser(product);
+            productExistValidator(stock);
+            enoughQuantityValidator(stock);
+            quantityZeroValidator();
+            addCart(stock);
+        }
+        return cart;
+    }
+
+    private void addCart(Stock stock) {
+        Product product = cart.getProductByName(name);
+
+        if (product != null) {
+            int beforeQuantity = cart.getProductByName(name).getQuantity();
+            cart.getProductByName(name).setQuantity(beforeQuantity + quantity);
+        }
+        if (product == null) {
+            cart.addProduct(new Product(name, stock.findPriceByName(name), quantity, "null"));
+        }
+
+    }
+
+    private void orderParser(String productText) {
+        try {
+            productText = productText.replace("[", "").replace("]", "");
+            name = productText.split("-")[0].trim();
+            quantity = Integer.parseInt(productText.split("-")[1].trim());
+        } catch (Exception e) {
```

Comment:
```suggestion
        } catch (ArrayIndexOutOfBoundsException | NumberFormatException e) {
```

에러를 명확하게 잡는게 차후에 오류를 잡는게 디버깅하는 데 도움이 될 거 같습니다! 다음 코드로 바꾸는게 어떨까요?

Reply:
`Exception`으로 너무 두루뭉실하게 잡았네요. 적용해볼게요!